docker/release/cudaq.Dockerfile

```diff
@@ -59,7 +59,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     && python3 -m pip install --no-cache-dir numpy \
     && ln -s /bin/python3 /bin/python
 
-ENV CPLUS_INCLUDE_PATH="$CPLUS_INCLUDE_PATH:/usr/include/c++/11/:/usr/include/x86_64-linux-gnu/c++/11"
+ENV CPLUS_INCLUDE_PATH="$CPLUS_INCLUDE_PATH:/usr/include/c++/12/:/usr/include/x86_64-linux-gnu/c++/12"
```

Comment:
Can we set the version in the variable somewhere it can be shared?

Reply:
Some references:
[CUDA compatibility](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#system-requirements)
[CUDA host compiler](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#supported-host-compilers)

Setting the CPLUS_INCLUDE_PATH to version 12 in the dev environment used to build CUDA Quantum results in a failure like [this](https://github.com/k2-fsa/k2/issues/972). See also the brief discussion [here](https://gitlab.kitware.com/cmake/cmake/-/issues/19458).

Setting the CUDAHOSTCXX to the same compiler we use to build CUDA Quantum results in the CUDA compiler not being found, since that version is generally not supported at this time. I have not figured out how to make CUDA try to compile with an officially not supported host compiler (I tried to set allow-unsupported-compiler but haven't been successful with it).